src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1050 @@
+            </Collapse>
+            <div style="padding-bottom:20px">
+              <Collapse
+                className="used-in-plans-collapse"
+                defaultExpanded={false}
+                title="Used in Plans"
+                tooltipContent="View plans this source is used in"
+              >
+                {#if selectedSourceLinkedDerivationGroupsPlans.length > 0}
+                  {#each selectedSourceLinkedDerivationGroupsPlans as linkedPlanDerivationGroup}
+                    <i>
+                      <a href="{base}/plans/{linkedPlanDerivationGroup.plan_id}">
+                        {$plans.find(plan => {
+                          return linkedPlanDerivationGroup.plan_id === plan.id;
+                        })?.name}
+                      </a></i
+                    >
+                  {/each}
+                {:else}
+                  <i class="st-typography-body">Not used in any plans</i>
+                {/if}
+              </Collapse>
+            </div>
+          </fieldset>
+
+          <button
+            class="st-button danger w-100"
+            style="margin-bottom:auto;"
+            use:permissionHandler={{
+              hasPermission: hasDeletePermission,
+              permissionError: deletePermissionError,
+            }}
+            on:click|stopPropagation={async () => {
+              if (selectedSource !== null) {
+                onDeleteExternalSource([selectedSource]);
+              }
+            }}
+          >
+            Delete external source
+          </button>
+        </div>
+      {:else}
+        <form
+          on:submit|preventDefault={onFormSubmit}
+          on:reset={() => {
+            parsed = undefined;
+            $createExternalSourceError = null;
+            $createExternalSourceTypeError = null;
+            $createDerivationGroupError = null;
+            $parsingError = null;
+            isDerivationGroupFieldDisabled = true;
+          }}
+        >
+          <AlertError class="m-2" error={$createExternalSourceError} />
+          <AlertError class="m-2" error={$createExternalSourceTypeError} />
+          <AlertError class="m-2" error={$createExternalEventTypeError} />
+          <AlertError class="m-2" error={$createDerivationGroupError} />
+          <AlertError class="m-2" error={$parsingError} />
+          <div class="file-upload-field">
+            <fieldset style:flex={1}>
+              <label for="file">Source File</label>
+              <input
+                class="w-100"
+                name="file"
+                required
+                type="file"
+                bind:files
+                use:permissionHandler={{
+                  hasPermission: hasCreatePermission,
+                  permissionError: createPermissionError,
+                }}
+              />
+            </fieldset>
+
+            <fieldset class="file-upload-fieldset">
+              {#if parsed}
+                <div style="padding-top:10px">
```

Comment:
We use a pixel grid of 8px so can you update your inline and class styles to use multiples of 8 or if that is too coarse multiples of 4?